File: backend/app/schemas/notification.py
from datetime import datetime

from pydantic import BaseModel


class NotificationCreate(BaseModel):
    title: str
    message: str
    category: str = "system"


class NotificationResponse(BaseModel):
    id: int
    title: str
    message: str
    category: str
    is_read: bool
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/types/notification.ts
export type NotificationCategory =
    | "weather"
    | "prediction"
    | "soil"
    | "system";

export interface Notification {
    id: number;
    title: string;
    message: string;
    category: NotificationCategory | string;
    is_read: boolean;
    created_at: string;
}
